packages/sources/view-starknet-latest-answer/src/transport/function.ts

```diff
@@ -0,0 +1,108 @@
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/function'
+import { AdapterInputError } from '@chainlink/external-adapter-framework/validation/error'
+import { RpcProvider } from 'starknet'
+
+const logger = makeLogger('View Starknet Latest Answer')
+
+export type StarknetLatestAnswerFunctionTransportTypes = BaseEndpointTypes
+
+type RequestParams = typeof inputParameters.validated
+
+export class StarknetLatestAnswerFunctionTransport extends SubscriptionTransport<StarknetLatestAnswerFunctionTransportTypes> {
+  provider: RpcProvider | undefined
+
+  async initialize(
+    dependencies: TransportDependencies<StarknetLatestAnswerFunctionTransportTypes>,
+    adapterSettings: StarknetLatestAnswerFunctionTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<StarknetLatestAnswerFunctionTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<StarknetLatestAnswerFunctionTransportTypes['Response']>
+    try {
+      response = await this._handleRequestStarknet(param)
+    } catch (e: unknown) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: (e as AdapterInputError)?.statusCode || 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequestStarknet(
+    param: RequestParams,
+  ): Promise<AdapterResponse<StarknetLatestAnswerFunctionTransportTypes['Response']>> {
+    const { address } = param
+
+    const networkEnvName = 'STARKNET_RPC_URL'
+
+    const rpcUrl = process.env[networkEnvName]
+
+    if (!rpcUrl) {
+      throw new AdapterInputError({
+        statusCode: 400,
+        message: `Missing '${networkEnvName}': '${rpcUrl}' environment variables.`,
+      })
+    }
+
+    if (!this.provider) {
+      this.provider = new RpcProvider({ nodeUrl: rpcUrl })
+    }
+
+    const callData = {
+      contractAddress: address,
+      entrypoint: 'latest_round_data',
+      calldata: [],
+    }
+
+    const providerDataRequestedUnixMs = Date.now()
+    // res: [round_id, answer, block_num, started_at, updated_at]
+    const res = await this.provider.callContract(callData)
```

Comment:
Are we able to check if this call succeeded? Would be useful to throw an explicit error if this fails (including the error returned by starknet.js)

Reply:
Added some handling